BE/src/main/java/com/team10/banchan/model/Category.java

```diff
@@ -0,0 +1,27 @@
+package com.team10.banchan.model;
+
+import org.springframework.data.annotation.Id;
+
+public class Category {
+
+    @Id
+    private final Long id;
+    private final String name;
+
+    Category(Long id, String name) {
+        this.id = id;
+        this.name = name;
+    }
+
+    public static Category newCategory(String name) {
+        return new Category(null, name);
+    }
```

Comment:
인자로 null을 넘겨주기 보다는 새로운 생성자를 정의해봅시다.

Reply:
https://docs.spring.io/spring-data/jdbc/docs/current/reference/html/#mapping.general-recommendations 에서,

> - Use factory methods instead of overloaded constructors to avoid @PersistenceConstructor — With an all-argument constructor needed for optimal performance, we usually want to expose more application use case specific constructors that omit things like auto-generated identifiers etc. It’s an established pattern to rather use static factory methods to expose these variants of the all-args constructor.

내부적으로 생성자가 여러개면, Spring Data JDBC에서 사용할 생성자에 `@PersistenceConstructor` 애노테이션을 달아줘야 합니다. 이를 피하기 위해서 팩토리 메소드를 사용하는 것을 추천한다고 하여, 추가적인 생성자를 정의하지 않고 팩토리 매소드에 id로 null을 사용했습니다.

추가로, https://docs.spring.io/spring-data/jdbc/docs/current/reference/html/#is-new-state-detection 에서,
id가 null 또는 0 이면, 새로운 Entity로 인식하여 Insert를 하고, 그렇지 않은 경우면 새 객체로 인식하지 않는다고 합니다.

Reply:
왜 애노테이션 붙이는 것을 피해야 할까요?
정답이 없는 문제지만, 저는 적당한 애노테이션의 사용은 좋지않나 생각해봅니다.
항상 K는 공식문서를 잘 읽으시는 것 같아요. 💯